tests/ui/arithmetic.stderr

```diff
@@ -1,127 +1,211 @@
    |     ^^^^^^^
 
 error: floating-point arithmetic detected
-  --> $DIR/arithmetic.rs:48:5
+  --> $DIR/arithmetic.rs:50:5
    |
 LL |     f * 2.0;
    |     ^^^^^^^
 
 error: floating-point arithmetic detected
-  --> $DIR/arithmetic.rs:49:5
+  --> $DIR/arithmetic.rs:51:5
    |
 LL |     f / 2.0;
    |     ^^^^^^^
 
 error: floating-point arithmetic detected
-  --> $DIR/arithmetic.rs:50:5
+  --> $DIR/arithmetic.rs:52:5
    |
 LL |     f - 2.0 * 4.2;
    |     ^^^^^^^^^^^^^
 
 error: floating-point arithmetic detected
-  --> $DIR/arithmetic.rs:51:5
+  --> $DIR/arithmetic.rs:53:5
    |
 LL |     -f;
    |     ^^
 
 error: floating-point arithmetic detected
-  --> $DIR/arithmetic.rs:53:5
+  --> $DIR/arithmetic.rs:55:5
    |
 LL |     f += 1.0;
    |     ^^^^^^^^
 
 error: floating-point arithmetic detected
-  --> $DIR/arithmetic.rs:54:5
+  --> $DIR/arithmetic.rs:56:5
    |
 LL |     f -= 1.0;
    |     ^^^^^^^^
 
 error: floating-point arithmetic detected
-  --> $DIR/arithmetic.rs:55:5
+  --> $DIR/arithmetic.rs:57:5
    |
 LL |     f *= 2.0;
    |     ^^^^^^^^
 
 error: floating-point arithmetic detected
-  --> $DIR/arithmetic.rs:56:5
+  --> $DIR/arithmetic.rs:58:5
    |
 LL |     f /= 2.0;
    |     ^^^^^^^^
 
-error: aborting due to 20 previous errors
+error: integer arithmetic detected
+  --> $DIR/arithmetic.rs:101:5
+   |
+LL |     3 + &1;
+   |     ^^^^^^
+
+error: integer arithmetic detected
+  --> $DIR/arithmetic.rs:102:5
+   |
+LL |     &3 + 1;
+   |     ^^^^^^
+
+error: integer arithmetic detected
+  --> $DIR/arithmetic.rs:103:5
+   |
+LL |     &3 + &1;
+   |     ^^^^^^^
+
+error: integer arithmetic detected
+  --> $DIR/arithmetic.rs:108:5
+   |
+LL |     a + x
+   |     ^^^^^
+
+error: integer arithmetic detected
+  --> $DIR/arithmetic.rs:112:5
+   |
+LL |     x + y
+   |     ^^^^^
+
+error: integer arithmetic detected
+  --> $DIR/arithmetic.rs:116:5
+   |
+LL |     x + y
+   |     ^^^^^
+
+error: integer arithmetic detected
+  --> $DIR/arithmetic.rs:120:5
+   |
+LL |     (&x + &y)
+   |     ^^^^^^^^^
+
+error: floating-point arithmetic detected
+  --> $DIR/arithmetic.rs:126:5
+   |
+LL |     3.1_f32 + &1.2_f32;
+   |     ^^^^^^^^^^^^^^^^^^
+
+error: floating-point arithmetic detected
+  --> $DIR/arithmetic.rs:127:5
+   |
+LL |     &3.4_f32 + 1.5_f32;
+   |     ^^^^^^^^^^^^^^^^^^
+
+error: floating-point arithmetic detected
+  --> $DIR/arithmetic.rs:128:5
+   |
+LL |     &3.5_f32 + &1.3_f32;
+   |     ^^^^^^^^^^^^^^^^^^^
+
+error: floating-point arithmetic detected
+  --> $DIR/arithmetic.rs:133:5
+   |
+LL |     a + f
+   |     ^^^^^
+
+error: floating-point arithmetic detected
+  --> $DIR/arithmetic.rs:137:5
+   |
+LL |     f1 + f2
+   |     ^^^^^^^
+
+error: floating-point arithmetic detected
+  --> $DIR/arithmetic.rs:141:5
+   |
+LL |     f1 + f2
+   |     ^^^^^^^
+
+error: floating-point arithmetic detected
+  --> $DIR/arithmetic.rs:145:5
+   |
+LL |     (&f1 + &f2)
+   |     ^^^^^^^^^^^
+
+error: aborting due to 34 previous errors
 
```

Comment:
Wait this should fail the `--limit-stderr-length` test 🤔

Reply:
I have a fix for this.

Reply:
Should I split up the into two, one for float and one for int arithmetic?

Reply:
Would be good. I just pushed the fix for this here. So please pull before splitting up the tests.